### PR TITLE
Add `:configurations` DSL for podspec dependencies.

### PR DESCRIPTION
**Pull request by segiddins**
Goes with https://github.com/CocoaPods/CocoaPods/pull/9149

### CODE REVIEW

File: lib/cocoapods-core/specification/dsl.rb [hunk 1]
```diff
@@ -676,6 +676,11 @@ def dependency(*args)
             end
           end
         end
+
+        options = version_requirements.last.is_a?(Hash) ? version_requirements.pop.dup : {}
+        whitelist = options.delete(:configurations)
+        raise Informative, "Unsupported options: #{options}" unless options.empty?
```
Comment (segiddins) [inline lib/cocoapods-core/specification/dsl.rb]: needs to be merged with the hash-handling logic below

Comment (dnkoutso) [reply lib/cocoapods-core/specification/dsl.rb]: may have a fix for this.

File: spec/specification/json_spec.rb [hunk 2]
```diff
@@ -86,11 +86,13 @@ module Pod
             'name' => 'GreenBanana',
             'source_files' => 'GreenBanana',
             'dependencies' => { 'AFNetworking' => [] },
+            'configuration_pod_whitelist' => { 'AFNetworking' => [] },
```
Comment (dnkoutso) [inline spec/specification/json_spec.rb]: @segiddins question from the original draft: It seems the original draft PR always set `attributes_hash['configuration_pod_whitelist']` whether any configurations were present or not. Are we OK with that or are unnecessarily "polluting" the JSON result?

Comment (dnkoutso) [reply spec/specification/json_spec.rb]: nm I think I was wrong, we were not setting this unless we had some whitelisted configs. Updating.

File: spec/specification/dsl_spec.rb [hunk 3]
```diff
@@ -268,6 +268,20 @@ module Pod
             @spec.dependency = 'JSONKit', '1.5'
           end.message.should.match /Cannot assign value to `dependency`. Did you mean: `dependency 'JSONKit', '1.5'`?/
         end
+
+        describe 'configurations for dependency' do
+          it 'stores the whitelisted configurations for the given dependency' do
+            @spec.dependency('SVProgressHUD', :configurations => %w(Debug Release))
```
Comment (segiddins) [inline spec/specification/dsl_spec.rb]: also maybe test with an empty array, and an array that includes a symbol?

Comment (dnkoutso) [reply spec/specification/dsl_spec.rb]: yes!

Comment (dnkoutso) [reply spec/specification/dsl_spec.rb]: done.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,11 @@
 
 ##### Enhancements
 
+* Add `:configurations` DSL for dependencies.
```
Comment (segiddins) [inline CHANGELOG.md]: nit: `podspec dependencies`

**Comment by dnkoutso** [comment]
Will merge on green.
